src/index.d.ts

```diff
@@ -0,0 +1,693 @@
+import { Script, TxOut } from 'bsv';
+
+declare interface CastTemplate {
+  lockingScript: {
+    script: unknown[];
+    size: ((params: object) => number) | number;
+    validate: (params: object) => void;
+  }
+}
+
+/**
+ * Forge transaction builder class.
+ */
+declare class Forge {
+  /**
+   * Instantiates a new Forge instance.
+   * 
+   * The accepted params are:
+   * 
+   * * `inputs` - list of input objects or cast instances
+   * * `outputs` - list of output objects or cast instances
+   * * `changeTo` - address to send change to
+   * * `changeScript` - bsv Script object to send change to
+   * * `options` - set `rates` or `debug` options
+   * 
+   * @param params Tx parameters
+   * @constructor
+   */
+  constructor(params?: Partial<{
+    inputs: object[] | Cast[];
+    outputs: object[] | Cast[];
+    changeTo: string;
+    changeScript: unknown;
+    nLockTime: number;
+    options: Partial<{
+      rates: unknown;
+      debug: unknown;
+    }>;
+  }>);
+
+  /**
+   * Returns the tx change address.
+   */
+  get changeTo(): string;
+
+  /**
+   * Sets the given address as the change address.
+   */
+  set changeTo(address: string);
+
+  /**
+   * Returns the tx nLockTime.
+   */
+  get nLockTime(): number;
+
+  /**
+   * Sets the given nLockTime on the tx.
+   * 
+   * If nLockTime < 500000000 it specifies the block number after which the tx
+   * can be included in a block. Otherwise it specifies UNIX timestamp after
+   * which it can be included in a block.
+   */
+  set nLockTime(lockTime: number);
+
+  /**
+   * The sum of all inputs.
+   */
+  get inputSum(): number;
+
+  /**
+   * The sum of all outputs.
+   */
+  get outputSum(): number;
+
+  /**
+   * Adds the given input to the tx.
+   * 
+   * The input should be a Cast instance, otherwise the given params will be
+   * used to instantiate a P2PKH Cast.
+   * 
+   * @param input Input Cast or P2PKH UTXO params
+   */
+  addInput(input: Cast[] | object[]): this;
+
+  /**
+   * Adds the given output params to the tx.
+   * 
+   * The params object should contain one of the following properties:
+   * 
+   * * `to` - Bitcoin address to create P2PKH output
+   * * `script` - hex encoded output script
+   * * `data` - array of chunks which will be automatically parsed into an OP_RETURN script
+   * 
+   * Unless the output is an OP_RETURN data output, the params must contain a
+   * `satoshis` property reflecting the number of satoshis to send.
+   * 
+   * For advanced use, Cast instances can be given as outputs. This allows
+   * sending to non-standard and custom scripts.
+   * 
+   * @param {Object} output Output params
+   */
+  addOutput(output: Partial<{
+    to: string;
+    script: string;
+    data: unknown[];
```

Comment:
`data` is an array where each element cab be either:

* string
* number
* OpCode instance (from bsv lib)
* Buffer

Reply:
Sorted